repository/maven.go

```diff
@@ -155,35 +148,66 @@ func (r *Maven) writeSettings() (path string, err error) {
 			path)
 		return
 	}
-	settings := id.Settings
-	settings, err = r.injectProxy(id)
+	defer f.Close()
+	id, found, err := r.findIdentity("maven")
 	if err != nil {
 		return
 	}
-	_, err = f.Write([]byte(settings))
+	settingsXML, err := mxj.NewMapXml([]byte(emptySettings))
+	if err != nil {
+		err = liberr.Wrap(err)
+		return "", err
+	}
+	if found {
+		addon.Activity(
+			"[MVN] Using credentials (id=%d) %s.",
+			id.ID,
+			id.Name)
+		// THis will use the settings in the idendity for settingsXML
+		settingsXML, err = r.injectProxy(id)
+		if err != nil {
+			return
+		}
+	}
+	settingsXML, err = r.injectCacheDir(settingsXML)
+	if err != nil {
+		return
+	}
+	settings, err := settingsXML.XmlIndent("", "  ")
+	if err != nil {
+		err = liberr.Wrap(err)
+		return "", err
+	}
+	_, err = f.Write(settings)
 	if err != nil {
 		err = liberr.Wrap(
 			err,
 			"path",
 			path)
 	}
-	_ = f.Close()
-	addon.Activity("[FILE] Created %s.", path)
+	addon.Activity("[FILE] Created %s. -- file: %s", path, settings)
 	return
 }
 
-//
+func (r *Maven) injectCacheDir(settings mxj.Map) (mxj.Map, error) {
+	err := settings.SetValueForPath(r.M2Dir, "settings.localRepository")
+	if err != nil {
+		err = liberr.Wrap(err)
+		return nil, err
+	}
+	return settings, nil
+}
+
 // injectProxy injects proxy settings.
-func (r *Maven) injectProxy(id *api.Identity) (s string, err error) {
-	s = id.Settings
-	m, err := mxj.NewMapXml([]byte(id.Settings))
+func (r *Maven) injectProxy(id *api.Identity) (mxj.Map, error) {
```

Comment:
All of the core projects use named return values by convention. Why did you change this?
Also, my previous comment requested the injectProxy() take a mxj.Map to be consistent with injectCacheDir() and because both the m2 dir and proxy settings should be set on either the emptySettings or the setting in the Identity.  Right?